### PR TITLE
Dataclean

### PR DESCRIPTION
**Pull request by AadyaRanjan**
Worked on Lasso and Ridge regression.

### CODE REVIEW

File: notebooks/Aadya/lasso_ridge.ipynb [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+    "# Scale the features\n",
+    "scaler = StandardScaler()\n",
+    "X_scaled = scaler.fit_transform(X)\n",
+    "\n",
+    "# Time series cross-validation\n",
+    "tscv = TimeSeriesSplit(n_splits=5)\n",
+    "\n",
+    "# Hyperparameter tuning for Lasso\n",
+    "lasso_params = {'alpha': [1e-7, 1e-6, 1e-5, 1e-4, 1e-2, 1e-1, 0, 0.1]}  # Wider range, smaller values\n",
+    "lasso_cv = GridSearchCV(Lasso(max_iter=10000), lasso_params, cv=tscv, scoring='neg_mean_squared_error')  # increase iterations\n",
```
Comment (virajparmaj) [inline notebooks/Aadya/lasso_ridge.ipynb:268]: `max_iter` could be increased to address convergence issues. adjusting alpha and checking standardization of features will help

File: notebooks/Aadya/lasso_ridge.ipynb [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+    "scaler = StandardScaler()\n",
+    "X_scaled = scaler.fit_transform(X)\n",
+    "\n",
+    "poly = PolynomialFeatures(degree=2, include_bias=False)\n",
+    "X_poly = poly.fit_transform(X_scaled)\n",
+    "\n",
+    "# Time series cross-validation\n",
+    "tscv = TimeSeriesSplit(n_splits=5)\n",
+    "\n",
+    "# Hyperparameter tuning for Lasso\n",
```
Comment (virajparmaj) [inline notebooks/Aadya/lasso_ridge.ipynb:83]: `lasso_params = {'alpha': np.logspace(-3, 1, 10)}
ridge_params = {'alpha': np.logspace(-3, 2, 10)}
`
these refined values might suit better

File: notebooks/Aadya/lasso_ridge.ipynb [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+    "y_pred_ridge = ridge_cv.best_estimator_.predict(X_poly)\n",
+    "\n",
+    "print(\"Improved Lasso Results:\")\n",
+    "print(f\"MSE: {mean_squared_error(y, y_pred_lasso)}\")\n",
+    "print(f\"R2: {r2_score(y, y_pred_lasso)}\")\n",
+    "\n",
+    "print(\"Improved Ridge Results:\")\n",
+    "print(f\"MSE: {mean_squared_error(y, y_pred_ridge)}\")\n",
+    "print(f\"R2: {r2_score(y, y_pred_ridge)}\")\n"
+   ]
```
Comment (virajparmaj) [inline notebooks/Aadya/lasso_ridge.ipynb:107]: for model evaluation , plotting predicted vs actual could be a better comparison

`plt.scatter(y, y_pred_lasso, label="Lasso Predictions", alpha=0.6)
plt.scatter(y, y_pred_ridge, label="Ridge Predictions", alpha=0.6)`

File: notebooks/Aadya/lasso_ridge.ipynb [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+    "\n",
+    "# Scale the features\n",
+    "scaler = StandardScaler()\n",
+    "X_scaled = scaler.fit_transform(X)\n",
+    "\n",
+    "# Time series cross-validation\n",
+    "tscv = TimeSeriesSplit(n_splits=5)\n",
+    "\n",
+    "# Hyperparameter tuning for Lasso\n",
+    "lasso_params = {'alpha': [1e-7, 1e-6, 1e-5, 1e-4, 1e-2, 1e-1, 0, 0.1]}  # Wider range, smaller values\n",
```
Comment (virajparmaj) [inline notebooks/Aadya/lasso_ridge.ipynb:267]: you can remove 'alpha=0' , it implies no regularization

File: notebooks/Aadya/lasso_ridge.ipynb [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+      "  model = cd_fast.enet_coordinate_descent(\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Best Lasso alpha: 0.1\n",
+      "Best Ridge alpha: 100\n",
+      "Improved Lasso Results:\n",
```
Comment (virajparmaj) [inline notebooks/Aadya/lasso_ridge.ipynb:163]: the values suggest overfitting, suggestions:

- Try PCA or removing redundant variables.
- Explore higher alphas for Ridge.
- Reduce the complexity to avoid overfitting.

**Comment by virajparmaj** [comment]
Overall seems fit, suggestions should solve the issue with warnings and other concerns. Great work!

File: notebooks/Aadya/lasso_ridge.ipynb [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+    "from sklearn.preprocessing import StandardScaler, PolynomialFeatures\n",
+    "from sklearn.linear_model import LinearRegression, Ridge\n",
+    "from sklearn.pipeline import make_pipeline\n",
+    "from sklearn.metrics import mean_squared_error, r2_score\n",
+    "from sklearn.model_selection import TimeSeriesSplit, GridSearchCV\n",
+    "\n",
+    "\n",
+    "\n",
+    "# Load the cleaned dataset\n",
+    "data = pd.read_csv(\"C:\\\\Users\\\\Aadya Ranjan\\\\Downloads\\\\merged_macroeconomic_credit.csv\")\n",
```
Comment (virajparmaj) [inline notebooks/Aadya/lasso_ridge.ipynb:61]: try using files within the folders, suppose the csv data is at data\merged_macroeconomic_credit.csv
this will help the collaborators run the notebooks without issues

**Review by virajparmaj — CHANGES_REQUESTED** [review]
